saltgui/static/scripts/routes/page.js

```diff
@@ -216,6 +217,127 @@ class PageRoute extends Route {
     container.appendChild(tr);
   }
 
+  _updateJobs(data) {
+    const jobContainer = this.getPageElement().querySelector(".jobs");
+    jobContainer.innerText = "";
+    const jobs = this._jobsToArray(data.return[0]);
+    this._sortJobs(jobs);
+
+    //Add seven most recent jobs
+    let shown = 0;
+    let i = 0;
+    while(shown < 7 && jobs[i] !== undefined) {
+      const job = jobs[i];
+      i = i + 1;
+      if(job.Function === "grains.append") continue;
+      if(job.Function === "grains.delkey") continue;
+      if(job.Function === "grains.delval") continue;
+      if(job.Function === "grains.items") continue;
+      if(job.Function === "grains.setval") continue;
+      if(job.Function === "pillar.items") continue;
+      if(job.Function === "pillar.obfuscate") continue;
+      if(job.Function === "runner.jobs.active") continue;
+      if(job.Function === "runner.jobs.list_job") continue;
+      if(job.Function === "runner.jobs.list_jobs") continue;
+      if(job.Function === "saltutil.find_job") continue;
+      if(job.Function === "saltutil.refresh_grains") continue;
+      if(job.Function === "saltutil.refresh_pillar") continue;
+      if(job.Function === "saltutil.running") continue;
+      if(job.Function === "schedule.delete") continue;
+      if(job.Function === "schedule.disable") continue;
+      if(job.Function === "schedule.disable_job") continue;
+      if(job.Function === "schedule.enable") continue;
+      if(job.Function === "schedule.enable_job") continue;
+      if(job.Function === "schedule.list") continue;
+      if(job.Function === "schedule.modify") continue;
+      if(job.Function === "schedule.run_job") continue;
+      if(job.Function === "sys.doc") continue;
+      if(job.Function === "wheel.config.values") continue;
+      if(job.Function === "wheel.key.accept") continue;
+      if(job.Function === "wheel.key.delete") continue;
+      if(job.Function === "wheel.key.list_all") continue;
+      if(job.Function === "wheel.key.reject") continue;
+
```

Comment:
This block is now duplicated. The part with the logic on which jobs to show (or not) should be separated.

Reply:
Sure. Will do this.